test/CAPI/aig.c

```diff
@@ -0,0 +1,177 @@
+//===- aig.c - Test for AIG C API ----------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+/* RUN: circt-capi-aig-test 2>&1 | FileCheck %s
+ */
+
+#include "circt-c/Dialect/AIG.h"
+#include "circt-c/Dialect/HW.h"
+#include "circt-c/Dialect/Seq.h"
+#include "mlir-c/BuiltinAttributes.h"
+#include "mlir-c/BuiltinTypes.h"
+#include "mlir-c/IR.h"
+
+#include <assert.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+void testAIGDialectRegistration(void) {
+  MlirContext ctx = mlirContextCreate();
+  mlirDialectHandleRegisterDialect(mlirGetDialectHandle__aig__(), ctx);
+
+  MlirDialect aig =
+      mlirContextGetOrLoadDialect(ctx, mlirStringRefCreateFromCString("aig"));
+  assert(!mlirDialectIsNull(aig));
```

Comment:
Could we use printf directly instead of relying on assertions? Assertions can be turned off with the right compiler flags and the test would pass trivially. Printf in the success case is more robust.
```suggestion
  if (!mlirDialectIsNull(aig));
    printf("success message");
```